packages/viewer/src/interactions/baseInteractionHandler.ts

```diff
@@ -80,6 +82,15 @@ export abstract class BaseInteractionHandler implements InteractionHandler {
       case 'pan':
         this.primaryInteraction = this.panInteraction;
         break;
+      case 'twist':
+        this.primaryInteraction = this.twistInteraction;
+        break;
```

Comment:
In the future, this class will become responsible for driving the behavior for these UI controls. One thing it'll probably handle is emitting of events that the UI control will listen to visually update the selected tool. By having twist in here and doing things like resets, the UI control is going to flicker between selected tools.

![Screen Shot 2021-02-25 at 11 07 22 AM](https://user-images.githubusercontent.com/37429/109190042-4ac33080-775a-11eb-90a6-eee171ae00f8.png)

I don't think this is the right place to handle twist interactions based on keyboard modifiers.

Reply:
What do you suggest then?

Reply:
Just thinking here: I could create a specific key down handler that then adds pointer event listeners. I don't know how we would do that and allow for seamless interactions between interactions though. Which is why this lives in the base interaction handler:

![one interaction](https://user-images.githubusercontent.com/49169871/109202420-94b31300-7768-11eb-95c7-dd31ad1bac7c.gif)